snapshot/linux/thread_snapshot_linux.cc

```diff
@@ -243,5 +279,20 @@ std::vector<const MemorySnapshot*> ThreadSnapshotLinux::ExtraMemory() const {
   return std::vector<const MemorySnapshot*>();
 }
 
+#ifdef CLIENT_STACKTRACES_ENABLED
+void ThreadSnapshotLinux::TrimStackTrace(uint64_t exception_address) {
+    auto start_frame = begin(frames_);
+    for (; start_frame != end(frames_); start_frame++) {
+      // These two addresses are never equivalent to each other 
+      if (start_frame->InstructionAddr() == exception_address) {
```

Comment:
need to revisit this later; this comparison is a little nonsensical right now as the `exception_address` and the instruction address values are in completely different integer ranges. unsure if it is because the exception address needs additional processing, or if libunwind's allocating a new address space mucks around with the original addresses, causing these two to not line up.

perhaps crashpad is doing something funny in linux in particular (eg the sigaltstack, maybe?) that makes it difficult to distinguish between the crashpad client frames and the actual exception frames.

Reply:
You were on the right track. Except you should have used `exception_->Context()->InstructionPointer()`, as that is the cpu context at the time of the crash (signal).

With that, things work just fine.